controllers/result.go

```diff
@@ -129,6 +133,20 @@ func (rc *ResultController) getResultID() (int64, error) {
 	return resultID, nil
 }
 
+func (rc *ResultController) queryFromParams() map[string]interface{} {
```

Comment:
As this function is building/genarating the query, how about rename it as `buildQueryFromParams` or `generateQueryFromParams`?